### PR TITLE
Feature/geode 3569

### PR DESCRIPTION
**Pull request by gesterzhou**
@upthewaterspout @jhuynh1 @boglesby 

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/i18n/LocalizedStrings.java [hunk 1]
```diff
@@ -7611,6 +7611,9 @@
   public static final StringId LuceneService_CANNOT_CREATE_INDEX_0_ON_REGION_1_WITH_ANALYZER_2_ON_FIELD_3_BECAUSE_ANOTHER_MEMBER_DEFINES_THE_SAME_INDEX_WITH_ANALYZER_4_ON_THAT_FIELD =
       new StringId(6631,
           "Cannot create Lucene index {0} on region {1} with analyzer {2} on field {3} because another member defines the same index with analyzer {4} on that field.");
+  public static final StringId LuceneService_CANNOT_CREATE_INDEX_0_ON_REGION_1_WITH_SERIALIZER_2_BECAUSE_ANOTHER_MEMBER_DEFINES_THE_SAME_INDEX_WITH_DIFFERENT_SERIALIZER_3 =
+      new StringId(6632,
+          "Cannot create Lucene index {0} on region {1} with Serializer {2} because another member defines the same index with different serializer {3}.");
```
Comment (jhuynh1) [inline geode-core/src/main/java/org/apache/geode/internal/i18n/LocalizedStrings.java]: nitpick, Serializer and serializer in the string should both be capitalized as "Serializer" or be "serializer", but probably not both

File: geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java [hunk 2]
```diff
@@ -36,18 +38,25 @@
...
+      Analyzer analyzer, Map<String, Analyzer> fieldAnalyzers, LuceneSerializer serializer) {
     this.indexName = indexName;
     this.regionPath = regionPath;
     this.fieldNames = fieldNames;
     this.analyzerClass = analyzer.getClass().getSimpleName();
     initializeFieldAnalyzers(fieldAnalyzers);
+    if (serializer == null) {
+      this.serializerClass = HeterogeneousLuceneSerializer.class.getSimpleName();
+    } else {
+      this.serializerClass = serializer.getClass().getSimpleName();
```
Comment (jhuynh1) [inline geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java]: Don't we want the full name and not the simple name?  What happens if there is a name collision...

File: geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java [hunk 3]
```diff
@@ -66,6 +75,10 @@ public String getAnalyzerClass() {
     return this.fieldAnalyzers;
   }
 
+  public String getSerializerClass() {
```
Comment (jhuynh1) [inline geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java]: Should this be named getSerializerClassName()?

File: geode-lucene/src/test/java/org/apache/geode/cache/lucene/test/LuceneTestUtilities.java [hunk 4]
```diff
@@ -79,6 +79,10 @@
       "Cannot create Lucene index index2 on region /region because it is not defined in another member.";
   public static final String CANNOT_CREATE_LUCENE_INDEX_DIFFERENT_INDEXES_3 =
       "Cannot create Lucene index index on region /region because it is not defined in another member.";
+  public static final String CANNOT_CREATE_LUCENE_INDEX_DIFFERENT_SERIALIZER =
+      "Cannot create Lucene index index on region /region with Serializer DummyLuceneSerializer because another member defines the same index with different serializer HeterogeneousLuceneSerializer.";
+  public static final String SERIALIZER_MATCHED =
+      "Serializer matched when creating Lucene index index on region /region.";
```
Comment (jhuynh1) [inline geode-lucene/src/test/java/org/apache/geode/cache/lucene/test/LuceneTestUtilities.java]: typo, index index

File: geode-core/src/main/java/org/apache/geode/internal/i18n/LocalizedStrings.java (hunk 1 above)
Comment (gesterzhou) [reply geode-core/src/main/java/org/apache/geode/internal/i18n/LocalizedStrings.java]: fixed

File: geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java (hunk 2 above)
Comment (gesterzhou) [reply geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java]: It follows the same idea of handling Analyzer class.

File: geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java (hunk 3 above)
Comment (gesterzhou) [reply geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/LuceneIndexCreationProfile.java]: We had a getAnalyzerClass() method. So follow the same idea.

File: geode-lucene/src/test/java/org/apache/geode/cache/lucene/test/LuceneTestUtilities.java (hunk 4 above)
Comment (gesterzhou) [reply geode-lucene/src/test/java/org/apache/geode/cache/lucene/test/LuceneTestUtilities.java]: This is not typo. The second "index" is the indexname.

File: geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneIndexCreationDUnitTest.java [hunk 5]
```diff
@@ -462,4 +474,29 @@ protected SerializableRunnableIF getAnalyzersIndexWithTwoFields2() {
...
+
+  protected SerializableRunnableIF getIndexWithDefaultSerializer() {
+    return () -> {
+      LuceneService luceneService = LuceneServiceProvider.get(getCache());
+      luceneService.createIndexFactory().setFields(new String[] {"field1", "field2"})
+          .create(INDEX_NAME, REGION_NAME);
+    };
+  }
+
+  protected SerializableRunnableIF getHeterogeneousLuceneSerializerCreationProfile() {
```
Comment (upthewaterspout) [inline geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneIndexCreationDUnitTest.java]: Is this method used in this test?

Comment (gesterzhou) [reply geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneIndexCreationDUnitTest.java]: I just added one more dunit test to use this method.

**Pull request closed without merging**